element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java

```diff
@@ -281,13 +302,23 @@ private Sender(String name) {
 
 		protected void work() throws InterruptedException, IOException {
 			RawData raw = outgoing.take(); // Blocking
+			/* check, if message should be sent with the "none correlation context" of UDP connector */
+			CorrelationContextMatcher correlationMatcher = getCorrelationContextMatcher();
+			if (null != correlationMatcher && !correlationMatcher.isToBeSent(raw.getCorrelationContext(), null)) {
+				if (LOGGER.isLoggable(Level.WARNING)) {
+					LOGGER.log(Level.WARNING, "UDPConnector ({0}) drops {1} bytes to {2}:{3}",
+							new Object[] { socket.getLocalSocketAddress(), datagram.getLength(), datagram.getAddress(),
+									datagram.getPort() });
+				}
+				return;
+			}
```

Comment:
We drop it.
And yes, introducing something as "Message.isDropped()" in the future would be helpfull. At least for unit tests :-).
But for this, it would easyly exceed the number of line.